framework/src/systems/NonlinearSystem.C

```diff
@@ -400,120 +416,99 @@ NonlinearSystem::converged()
 void
 NonlinearSystem::computeScalingJacobian()
 {
-#ifdef LIBMESH_HAVE_PETSC
-  if (dynamic_cast<PetscMatrix<Real> *>(_transient_sys.matrix))
-  {
-#if !PETSC_VERSION_LESS_THAN(3, 9, 0)
-    _console << "\nPerforming automatic scaling calculation\n\n";
-
-    TIME_SECTION(_compute_scaling_jacobian_timer);
-
-    auto & petsc_matrix = *static_cast<PetscMatrix<Real> *>(_transient_sys.matrix);
+  _console << "\nPerforming automatic scaling calculation\n\n";
 
-    if (!petsc_matrix.local_m())
-      mooseError("MOOSE doesn't currently support automatic scaling when there are any processes "
-                 "owning zero dofs. Check back soon :-)");
+  TIME_SECTION(_compute_scaling_jacobian_timer);
 
-    _computing_scaling_jacobian = true;
-    _fe_problem.computeJacobianSys(_transient_sys, *_current_solution, *_transient_sys.matrix);
-    _computing_scaling_jacobian = false;
+  mooseAssert(_scaling_matrix_tag != Moose::INVALID_TAG_ID,
+              "The scaling matrix has not been created. There must have been an issue in "
+              "initialization of the NonlinearSystem");
```

Comment:
Instead of an ID, I think we should use a tag name, e.g., "scaling_matrix". That way will be easier to understand the code. Then you can check if the tag name exists here.

Reply:
I don't see any APIs like `hasMatrix(std::string name)` or `getMatrix(std::string name)`. Only the `hasMatrix (TagID tag)` and `getMatrix (TagID tag)` versions. Do I have to add those?

Reply:
If we do not have, we could add one. Not hard:)